### DISCUSSION TITLE
Build DynamicDataType from ROS

### DISCUSSION THREAD

**Discussion by cedricmelancon** [eProsima/Fast-DDS#3725] [Q&A]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I am using a rosbag play from ROS2 Foxy and I would like to create a FastDDS subscriber to a ROS topic, but I don't have the type definitions. Is it possible to use Discovery to build a DynamicDataType? I tried to created a TypeLookup, but I get no call for on_type_discovery. I do get calls for on_publisher_discovery.

Why don't I get type discovery calls?

### Current behavior

No type discovery called

### Steps to reproduce

N/A

### Fast DDS version/commit

2.10.1

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

UDPv4, TCPv4

**Comment by JLBuenoLopez** [comment]
Thanks for your contribution @cedricmelancon 

TypeLookup Service cannot be configured in ROS 2 Foxy. The TypeLookup Service is enabled at the time of DomainParticipant creation and is a non-mutable QoS. ROS 2 does not expose the DomainParticipant creation nor allows to configure its QoS. The only possibility is configure the QoS using a XML configuration file, but TypeLookup Service XML configuration has just recently been supported (in [Fast DDS v2.11.0](https://github.com/eProsima/Fast-DDS/releases/tag/v2.11.0)). ROS 2 Foxy uses Fast DDS v2.1.x which is no longer supported.

You might consider using a native Fast DDS application instead of a ROS 2 application. However, TypeLookup Service has several known issues that are now being dealt with. The Service is known to work only with the simplest types but when dependencies or complex types are used, the behavior is quite unstable.It is expected that the next Fast DDS minor release includes an important refactor of the TypeLookup Service. How this feature is going to be exposed in ROS 2 is still to be defined.
